ParameterCodec: make encoding symmetric to decoding

sttts opened the pull request:
Consider all returned `ObjectKinds` during parameter decoding. Before this PR only the first (arbitrary returned) gvk was used to skip conversion.

sttts committed:
ParameterCodec: make encoding symmetric to decoding

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/43131)
<!-- Reviewable:end -->

smarterclayton commented:
Add a simple test or two and then lgtm.  There should be one stressing scheme already (synthetic schemes with types)

ixdy committed:
Use munged semantic version for side-loaded docker tag

Commit:
Remove incorrect and unused function

The newClusterClientset() has insteaded of GetClientsetForCluster(),
and GetClientsetForCluster() run wrong. Let's remove it.

sttts committed:
Add EncodeParameters test

grodrigues3 commented:
Looks like this isn't going in to the 1.7 milestone.  No lgtm prior to code freeze

spiffxp commented:
/remove-lifecycle frozen
/lifecycle rotten
/close
I'm closing this due to inactivity. Please /reopen if you wish to continue working on this
